### PR TITLE
feat: add BIGNUMERIC support

### PR DESCRIPTION
**Pull request by stephaniewang526**
Expose BIGNUMERIC type in the client 

cc: @mingyuzhong

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardSQLTypeName.java [hunk 1]
```diff
@@ -32,6 +32,8 @@
   FLOAT64,
   /** A decimal value with 38 digits of precision and 9 digits of scale. */
   NUMERIC,
+  /** A decimal value with 38 digits of precision and 38 digits of scale. */
```
[the remark this answers is not in the document]
Comment (stephaniewang526) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardSQLTypeName.java]: done

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/QueryParameterValueTest.java [hunk 2]
```diff
@@ -141,6 +141,31 @@ public void testNumeric() {
...
+  @Test
+  public void testBigNumeric() {
+    QueryParameterValue value =
+        QueryParameterValue.bigNumeric(new BigDecimal("0.33333333333333333333333333333333333333"));
+    QueryParameterValue value1 =
+        QueryParameterValue.bigNumeric(new BigDecimal("0.50000000000000000000000000000000000000"));
+    QueryParameterValue value2 =
+        QueryParameterValue.bigNumeric(new BigDecimal("0.00000000500000000000000000000000000000"));
+    QueryParameterValue value3 =
+        QueryParameterValue.bigNumeric(new BigDecimal("-0.00000000500000000000000000000000000000"));
```
[the remark this answers is not in the document]
Comment (stephaniewang526) [reply google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/QueryParameterValueTest.java:153]: done

**Comment by stephaniewang526** [comment]
Cannot merge until BigQuery backend is ready.

**Review by pmakani — APPROVED** [review]

**Comment by shollyman** [comment]
Modulo any updates bringing this PR back in line with master, this should be good to merge now.

**Review by shollyman — APPROVED** [review]

**Pull request merged**